### ISSUE TITLE
`expect().toContain()` error message doesn't show expected string

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/deno_std#4659]
**Describe the bug**

When an `expect().toContain()` assertion failed we get a very generic error message:

```txt
error: AssertionError: The value doesn't contain the expected item
      throw new AssertionError("The value doesn't contain the expected item");
```

This makes it difficult to know what exactly went wrong.

**Steps to Reproduce**

```ts
import { expect } from "jsr:@std/expect";

Deno.test("foo", () => {
  expect("foobar").toContain("baz");
});
```

**Expected behavior**

Print something like `Input string did not contain "baz"`. Not sure what to do for long strings though.

**Environment**

- OS: macOS 14.4.1
- deno version: 1.42.4
- std version: 0.224.0

**Comment by JakeAve** [comment]
I've never used `toContainEqual` but it looks like we could do the same thing there. Should I include that in the PR?

**Comment by marvinhagemeister** [comment]
Good idea, makes sense to add this to `toContainEqual` too

**Comment by kt3k** [comment]
fixed in #4750
